packages/integration-tests/local.js

```diff
@@ -0,0 +1,149 @@
+  return Promise.all(
+    workflow.steps.map(
+      (step) => fs.remove(path.join(step.lambda, cmaFolder))
+    )
+  );
+}
+
+/**
+ * Build a cumulus message for a given workflow
+ *
+ * @param {Object} workflow - a test workflow object
+ * @param {Object} configOverride - a cumulus config override object
+ * @param {Array} cfOutputs - mocked outputs of a CloudFormation template
+ * @returns {Object} the generated cumulus message
+ */
+function messageBuilder(workflow, configOverride, cfOutputs) {
+  const workflowConfigs = {}
+  workflow.steps.forEach((step) => {
+    workflowConfigs[step.name] = step.cumulusConfig;
+  });
+
+  const config = {
+    stack: 'somestack', 
+    workflowConfigs: {
+      [workflow.name]: workflowConfigs
+    }
+  };
+  Object.assign(config, configOverride);
+  config.stackName = config.stack;
+
+  const message = template(workflow.name, { States: workflowConfigs }, config, cfOutputs);
+  message.cumulus_meta.message_source = 'local';
+  return message;
+}
+
+/**
+ * Runs a given workflow step (task)
+ *
+ * @param {string} lambdaPath - the local path to the task (e.g. path/to/task)
+ * @param {string} lambdaHandler - the lambda handler (e.g. index.hanlder) 
+ * @param {Object} message - the cumulus message input for the task
+ * @param {string} stepName - name of the step/task
+ * @returns {Promise.<Object>} the cumulus message returned by the task
+ */
+async function runStep(lambdaPath, lambdaHandler, message, stepName) {
+  const taskFullPath = path.join(process.cwd(), lambdaPath);
+  const src = path.join(taskFullPath, 'adapter.zip');
+  const dest = path.join(taskFullPath, 'cumulus-message-adapter');
+  let resp;
+
+  process.env.CUMULUS_MESSAGE_ADAPTER_DIR = dest;
+
+  // add step name to the message
+  message.cumulus_meta.task = stepName;
+
+  try {
+    // add message adapter to task folder
+
+    // run the task
+    const moduleFn = lambdaHandler.split('.');
+    const moduleFileName = moduleFn[0];
+    const moduleFunctionName = moduleFn[1];
+    const task = require(`${taskFullPath}/${moduleFileName}`);
+
+    console.log(`Started execution of ${stepName}`);
+
+    return new Promise((resolve, reject) => {
+      task[moduleFunctionName](message, {}, (e, r) => {
+        if (e) return reject(e);
+        console.log(`Completed execution of ${stepName}`);
+        return resolve(r);
+      });
+    });
+  }
+  finally {
+    await fs.remove(src);
+  }
+}
+
+/**
+ * Executes a given workflow by running each step in the workflow
+ * one after each other
+ *
+ * @param {Object} workflow - a test workflow object
+ * @param {Object} message - input message to the workflow
+ * @returns {Promise.<Object>} an object that includes the workflow input/output
+ *  plus the output of every step
+ */
+async function runWorkflow(workflow, message) {
+  const trail = {
+    input: clone(message),
+    stepOutputs: {},
+    output: {}
+  };
+
+  let stepInput = clone(message);
+
+  for (const step of workflow.steps) {
```

Comment:
Curious - is there a reason to use this iteration syntax over `workflowSteps.forEach((step) => {` ?

Reply:
This has to happen in sequence. If we don't use for of, we have to use some other library to ensure each async task happen after the other one ended.

Reply:
I see that makes sense